app/views/components/_markdown-editor.html.erb

```diff
@@ -1,17 +1,26 @@
+<%
+  textarea||= {}
+  edit_button_text||= 'Edit'
+  preview_button_text||= 'Preview'
+%>
 <div class="app-c-markdown-editor" data-module="markdown-editor">
+  <% if textarea[:label] %>
   <%= render "govuk_publishing_components/components/label", {
       text: textarea[:label][:text],
       html_for: textarea[:id]
   } %>
   <% textarea[:label] = nil %>
-  <div class="app-c-markdown-editor__preview-toggle">
-    <button class="app-c-markdown-editor__button app-c-markdown-editor__button--edit js-edit-button">Edit markdown</button>
-    <button class="app-c-markdown-editor__button app-c-markdown-editor__button--preview js-preview-button">Preview markdown</button>
+  <% end %>
+  <div class="app-c-markdown-editor__preview-toggle js-markdown-toggle-bar">
+    <button class="app-c-markdown-editor__button app-c-markdown-editor__button--edit js-markdown-edit-button"><%= edit_button_text %></button>
+    <button class="app-c-markdown-editor__button app-c-markdown-editor__button--preview js-markdown-preview-button"><%= preview_button_text %></button>
   </div>
-  <div class="app-c-markdown-editor__input js-editor-input">
+  <div class="app-c-markdown-editor__input js-markdown-editor-input">
+    <% if textarea[:name] %>
```

Comment:
Can you explain what this new conditional is for?

Reply:
The minimum we need to render a textarea component is its name, hence we make sure we have one set before trying to render it

Reply:
Hmm...this sounds like a validation in disguise to me. What would be the impact of not having this conditional, assuming people use the component as documented in the manual?

Reply:
The textarea component will complain (error) about not having a name

Reply:
What I'm getting at is this: is there a use case for this component where we don't specify a name?

Reply:
Removed this condition.